Tasks/DotNetCoreInstallerV1/dotnetcoreinstaller.ts

```diff
@@ -0,0 +1,76 @@
+import * as path from 'path';
+
+import * as tl from 'vsts-task-lib/task';
+import * as toolLib from 'vsts-task-tool-lib/tool';
+import { DotNetCoreVersionFetcher, VersionInfo } from "./versionfetcher";
+import { VersionInstaller } from "./versioninstaller";
+import { VersionParts, Constants } from "./versionutilities";
+
+async function run() {
+    let packageType = tl.getInput('packageType', true).toLowerCase();
+    let versionSpec = tl.getInput('version', true);
+    let installationPath = tl.getInput('installationPath', false);
+    if (!installationPath) {
+        installationPath = path.join(tl.getVariable('Agent.ToolsDirectory'), "dotnet");
+    }
+    let includePreviewVersions: boolean = tl.getBoolInput('includePreviewVersions', false) || false;
+
+    console.log(tl.loc("ToolToInstall", packageType, versionSpec));
+    VersionParts.ValidateVersionSpec(versionSpec);
+
+    let versionFetcher = new DotNetCoreVersionFetcher();
+    let versionInfo: VersionInfo = await versionFetcher.getVersionInfo(versionSpec, packageType, includePreviewVersions);
+    if (!versionInfo) {
+        throw tl.loc("MatchingVersionNotFound", versionSpec);
+    }
+
+    let dotNetCoreInstaller = new VersionInstaller(packageType, installationPath);
+    if (!dotNetCoreInstaller.isVersionInstalled(versionInfo.version)) {
+        await dotNetCoreInstaller.downloadAndInstall(versionInfo, versionFetcher.getDownloadUrl(versionInfo, packageType));
+    }
+
+    toolLib.prependPath(installationPath);
+
+    // By default disable Multi Level Lookup unless user wants it enabled.
+    let restrictMultiLevelLookup = true;
+    try {
+        restrictMultiLevelLookup = tl.getBoolInput("restrictMultiLevelLookup", true);
```

Comment:
why will it throw.

Reply:
It is not a required input, so if someone doesn't specify this input in yaml file. And due to tl.getboolinput's behaviour of always sending false as default (even in case input is missing), i am using getboolinput with required = true. so that if it throws, we can default to true value and disable multilevel lookup

Reply:
Cant you achieve this with the default value of true

Reply:
The library function is not exposing a function that returns back a default value of choice. So to know if the value is actually present or not I am passing required=true, so that it fails in case input is not define and then use default value.